crates/in_stream/src/tcp.rs

```diff
@@ -122,10 +124,12 @@ impl InStreamTcp {
 
     fn priv_new(
         stream: std::net::TcpStream,
+        url: Url2,
         connecting: Option<TcpConnectingData>,
     ) -> Result<Self> {
         Ok(Self {
             stream,
+            url,
```

Comment:
priv_new looks liike it will always returns OK, do we need the result here?

Reply:
The places `priv_new` is called from expect results - this saves me putting `Ok()`s in those spots : )